src/main/java/core/basesyntax/service/serviceimpl/FruitStringsCheckImpl.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.service.serviceimpl;
+
+import core.basesyntax.exception.DataFileCorrupted;
+import core.basesyntax.service.interfaces.FruitStringsCheck;
+import java.util.List;
+
+public class FruitStringsCheckImpl implements FruitStringsCheck {
+    private static final int EXPECTED_COLUMN_COUNT = 3;
+
+    @Override
+    public boolean checkFruitQuantity(List<String> rawStrings) {
+        if (rawStrings == null || rawStrings.isEmpty()) {
+            return false;
+        }
+        String headerLine = rawStrings.get(0);
+        String[] headerColumns = headerLine.split(",");
+        if (headerColumns.length != EXPECTED_COLUMN_COUNT) {
+            throw new DataFileCorrupted("Header row is missing");
+        }
+        for (String line : rawStrings.subList(1, rawStrings.size())) {
+            String[] columns = line.split(",");
+            if (columns.length != EXPECTED_COLUMN_COUNT) {
+                throw new DataFileCorrupted("Data rows is missing, expected "
+                + EXPECTED_COLUMN_COUNT
+                + " but only "
+                + columns.length
+                + " is present");
+            }
+            for (String column : columns) {
+                if (column == null || column.trim().isEmpty()) {
+                    throw new DataFileCorrupted("Data in column is absent");
+                }
+            }
+        }
+        return true;
```

Comment:
all of that is so unnecessary, would have been enough to make a private method in the transaction parser where you check the length of the list and that is it
There are cases when we can bring out the validation logic outside but we would name it like `RawDataValidator` with method `validate()` and your class and method names just do not represent what it actually does, or are not completely understandable